src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/AccessorBuilder.cs

```diff
@@ -0,0 +1,153 @@
+            return make;
+        }
+
+        public static Getter BuildGetAccessor(MemberInfo memberInfo)
+        {
+            if (memberInfo is PropertyInfo)
+            {
+                var propInfo = (PropertyInfo)memberInfo;
+                Func<PropertyInfo, Getter> buildGetAccessorGeneric = (Func<PropertyInfo, Getter>)s_buildGetAccessorInternal.MakeGenericMethod(propInfo.DeclaringType, propInfo.PropertyType).CreateDelegate(typeof(Func<PropertyInfo, Getter>));
+                Getter accessor = buildGetAccessorGeneric(propInfo);
+                return accessor;
+            }
+            else if (memberInfo is FieldInfo)
+            {
+                return (obj) =>
+                {
+                    FieldInfo fieldInfo = (FieldInfo)memberInfo;
+                    var value = fieldInfo.GetValue(obj);
+                    return value;
+                };
+            }
+            else
+            {
+                throw new InvalidOperationException($"The type, {memberInfo.GetType()}, of memberInfo is not supported.");
+            }        
+        }
+
+        public static Setter BuildSetAccessor(MemberInfo memberInfo)
+        {
+            if (memberInfo is PropertyInfo)
+            {
+                PropertyInfo propInfo = (PropertyInfo)memberInfo;
+                if (propInfo.CanWrite)
+                {
+                    Func<PropertyInfo, Setter> buildSetAccessorGeneric = (Func<PropertyInfo, Setter>)s_buildSetAccessorInternal.MakeGenericMethod(propInfo.DeclaringType, propInfo.PropertyType).CreateDelegate(typeof(Func<PropertyInfo, Setter>));
+                    Setter accessor = buildSetAccessorGeneric(propInfo);
+                    return accessor;
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Property {propInfo.Name} of type {DataContract.GetClrTypeFullName(propInfo.DeclaringType)} cannot be set.");
+                }
+            }
+            else if (memberInfo is FieldInfo)
+            {
+                FieldInfo fieldInfo = (FieldInfo)memberInfo;
+                return (ref object obj, object val) =>
+                {
+                    fieldInfo.SetValue(obj, val);
+                };
+            }
+            else
+            {
+                throw new NotImplementedException("Unknown member type");
+            }
+        }
+
+        private static object Make<T>() where T : new()
+        {
+            var t = new T();
+            return t;
+        }
+
+        private static Getter BuildGetAccessorInternal<DeclaringType, PropertyType>(PropertyInfo propInfo)
+        {
+            Func<DeclaringType, PropertyType> getMethod = propInfo.GetMethod.CreateDelegate<Func<DeclaringType, PropertyType>>();
+
+            return (obj) =>
+            {
+                return getMethod((DeclaringType)obj);
```

Comment:
Closures are about binding parameters via an implicit state object that gets passed into the function via a hidden parameter. Type information is about code generation so the Type won't get passed in.